tezos-encoding/src/nom.rs

```diff
@@ -265,14 +265,28 @@ hash_nom_reader!(PublicKeyP256);
 hash_nom_reader!(PublicKeyBls);
 hash_nom_reader!(SecretKeyEd25519);
 hash_nom_reader!(SecretKeyBls);
-hash_nom_reader!(Signature);
+hash_nom_reader!(UnknownSignature);
 hash_nom_reader!(Ed25519Signature);
 hash_nom_reader!(Secp256k1Signature);
 hash_nom_reader!(P256Signature);
 hash_nom_reader!(BlsSignature);
 hash_nom_reader!(NonceHash);
 hash_nom_reader!(SmartRollupHash);
 
+impl<'a> NomReader<'a> for crypto::signature::Signature {
+    fn nom_read(input: &'a [u8]) -> NomResult<'a, Self> {
+        let (rest, v) = dynamic(bytes)(input)?;
+        if let Ok(v) = Self::from_vec(v) {
+            Ok((rest, v))
+        } else {
+            Err(Err::Error(DecodeError::limit(
+                input,
+                BoundedEncodingKind::Bounded,
+            )))
```

Comment:
This error seems a bit dubious, doesn't it? `limit(Bounded)` is returned from `bounded` parser, which happens when the second (data) parser reaches EOF. This isn't quite what's happening here. Perhaps it would be better to add a new `BoundedEncodingKind::Signature`?

Reply:
Makes sense, introduced a `Signature` variant.